sdk/keyvault/azure-security-keyvault-jca/src/main/java/com/azure/security/keyvault/jca/implementation/utils/JsonConverterUtil.java

```diff
@@ -3,61 +3,84 @@
 
 package com.azure.security.keyvault.jca.implementation.utils;
 
-import com.fasterxml.jackson.core.JsonProcessingException;
-import com.fasterxml.jackson.databind.DeserializationFeature;
-import com.fasterxml.jackson.databind.ObjectMapper;
+import com.azure.json.JsonProviders;
+import com.azure.json.JsonReader;
+import com.azure.json.JsonSerializable;
+import com.azure.json.JsonWriter;
 
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
 import java.util.logging.Logger;
 
 import static java.util.logging.Level.WARNING;
 
 /**
- * The Jackson JsonConverter.
+ * The JSON Converter.
  */
 public final class JsonConverterUtil {
-    private static final ObjectMapper FROM_JSON_MAPPER = new ObjectMapper()
-        .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
-    private static final ObjectMapper TO_JSON_MAPPER = new ObjectMapper();
-
     /**
      * Stores the logger.
      */
     private static final Logger LOGGER = Logger.getLogger(JsonConverterUtil.class.getName());
 
     /**
-     * From JSON.
+     * Deserializes the {@code json} as an instance of {@link JsonSerializable}.
+     *
+     * @param jsonSerializable The {@link JsonSerializable} represented by the {@code json}.
+     * @param json The JSON being deserialized.
      *
-     * @param string the string.
-     * @param resultClass the result class.
-     * @return the object, or null if the conversion failed.
+     * @return An instance of {@code jsonSerializable} based on the {@code json}.
+     *
+     * @throws IOException If an error occurs during deserialization.
+     * @throws IllegalStateException If the {@code jsonSerializable} does not have a static {@code fromJson} method.
+     * @throws Error If an error occurs during deserialization.
      */
-    public static Object fromJson(String string, Class<?> resultClass) {
-        LOGGER.entering("JsonConverterUtil", "fromJson", new Object[] { string, resultClass });
-        Object result = null;
-        try {
-            result = FROM_JSON_MAPPER.readValue(string, resultClass);
-        } catch (JsonProcessingException e) {
-            LOGGER.log(WARNING, "Unable to convert from JSON", e);
+    @SuppressWarnings("unchecked")
+    public static <T extends JsonSerializable<T>> T fromJson(Class<T> jsonSerializable, String json)
```

Comment:
Slight modification to this class to complete remove the need for reflection here.

```java
public static <T extends JsonSerializable<T>> T fromJson(ReadValueCallback<JsonReader, T> deserializationFunction, String json) throws IOException {
    try (JsonReader jsonReader = JsonProviders.createReader(json)) {
        return deserializationFunction(jsonReader);
    }
}
```

then methods call this can change to

```java
AccessToken accessToken = JsonConverterUtil.fromJson(AccessToken::fromJson, jsonString);
```